pytissueoptics/scene/scene/scene.py

```diff
@@ -1,8 +1,60 @@
-from typing import List
+from typing import List, Dict
 
+from pytissueoptics.scene.materials import Material
+from pytissueoptics.scene.geometry import Vector
 from pytissueoptics.scene.solids import Solid
 
 
 class Scene:
-    def __init__(self, solids: List[Solid]):
-        self.solids = solids
+    def __init__(self, ignoreIntersections=False):
+        self._solids = []
+        self._ignoreIntersections = ignoreIntersections
+
+    def add(self, solid: Solid, position: Vector = None):
+        if position:
+            solid.translateTo(position)
+        if not self._ignoreIntersections:
+            self._validate(solid)
+        self._solids.append(solid)
+
+    def _validate(self, newSolid: Solid):
+        """ Assert newSolid position is valid and make proper adjustments so that the
+        material at each solid interface is well defined. """
+        if len(self._solids) == 0:
+            return
+
+        intersectingSuspects = self._findIntersectingSuspectsFor(newSolid)
+        if len(intersectingSuspects) == 0:
+            return
+
+        intersectingSuspects.sort(key=lambda s: s.getBoundingBox().xMax - s.getBoundingBox().xMin, reverse=True)
+
+        solidUpdates: Dict[Solid, Material] = {}
+        for otherSolid in intersectingSuspects:
+            if newSolid.contains(*otherSolid.getVertices()):
+                self._assertIsNotAStack(newSolid)
+                solidUpdates[otherSolid] = newSolid.getMaterial()
+                break
+            elif otherSolid.contains(*newSolid.getVertices()):
+                self._assertIsNotAStack(otherSolid)
+                solidUpdates[newSolid] = otherSolid.getMaterial()
+            else:
+                raise NotImplementedError("Cannot place a solid that partially intersects with an existing solid. ")
+
+        for (solid, material) in solidUpdates.items():
+            solid.setOutsideMaterial(material)
+
+    def _findIntersectingSuspectsFor(self, solid) -> List[Solid]:
+        solidBBox = solid.getBoundingBox()
+        intersectingSuspects = self._solids
+        for axis in range(3):
+            for suspect in intersectingSuspects:
+                suspectBBox = suspect.getBoundingBox()
+                if suspectBBox[axis][0] > solidBBox[axis][1] or suspectBBox[axis][1] < solidBBox[axis][0]:
```

Comment:
could be made into a method in the `BoundingBox` class.
BoundingBox.intersects(other) -> bool:

Reply:
Sorry I was forgetting why I did this in the first place. The algo looks at one axis at a time. If a suspect doesn't intersect for a given axis, we remove it from the suspects list. This allows for almost a 3x decrease in time complexity on average. 

I see 3 options:
- Leave it this way. Good performance but not very TDA friendly. 
- Create bbox.intersectsOnAxis() and keep same algorithm, but this method might sound too specific from the bbox point of view. 
- Create bbox.intersects(other). Maybe 2.5 times slower on average but respects TDA. And there is a case to say that we don't need performance for this method since it is only called when adding a solid to the scene, scales with the number of solids, and we might not have much solid in the scene (I don't know if this remains the case when we load complex obj file).

Reply:
I'd agree more with the third option, which is what I was thinking in the first place.
If it is kept this way, groupObject will have its own bbox, so it shall not be a problem, if it still works this way in the future.